### PR TITLE
[Do not merge] Adding workspace level quota management APIs

### PR DESCRIPTION
**Pull request by leizhaoPS**
To update quota, we use the Post call with the action quota URI (similar to Usages): subscriptions/{subscriptionId}/providers/Microsoft.BatchAI/locations/{locationName}/updateQuotas

Parameters of this call:
Subscription ID (uri)
location (uri)
vmFamily (body)
workspaceId (body)
Quota (body)
api-version (query)

To get quota, we use Post call on the list action quota URI: subscriptions/{subscriptionId}/providers/Microsoft.BatchAI/locations/{locationName}/listQuotas

Parameters of this call:
Subscription ID (uri)
location (uri)
api-version (query)
Changes:
Added new API for update Quotas
Added new API for list Quotas
Added two examples:
ListWorkspaceQuotasByVMFamily.json
UpdateQuota.json

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (nschonni) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]:
```suggestion
        "vmFamily": {
```

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (nschonni) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]:
```suggestion
        "amlWorkspaceName": {
```

**Comment by leizhaoPS** [comment]
Consider folks will give comments, close the draft PR and I will reopen a new PR

**Comment by leizhaoPS** [comment]
reopen as normal PR

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: Thanks for the comments, I will update those soon!

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (leizhaoPS) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: typo: update quota

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (leizhaoPS) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: typo: List workspace quotas by VMFamily

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: POST apis should start with a verb, i.e. `updateQuotas`

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: In order to independently RBAC read and write, this should be modeled as a second POST api: `listQuotas` - It could be a GET with an API name of: `quotas` but that would also imply that the response body follows standard proxy resource schema (id, type, name, properties). For response body consistency, independent POSTs can be used. You can map the authZ check for the listQuotas to a READ action in the manifest

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (KrisBash) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: Highly recommend putting these new APIs in a preview version for initial release

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: Thanks for the feedback, I am really appreciated it! I will modify those and update to preview.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json
Comment (nschonni) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]:
```suggestion
          "description": "Fully qualified resource identifier of the workspace."
```

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json
Comment (leizhaoPS) [reply specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/stable/2018-11-19/machineLearningServices.json]: I moved the code to preview

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json
Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: >updateQuotas [](start = 102, length = 12)

I believe this should be modelled as PUT  /quotas/{quotaName} instead.. 
and GET /quotas for collection call.. that is the pattern that all RPs used.

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json
Comment (sanjaiganesh) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2018-03-01-preview/machineLearningServices.json]: >Quotas_ListByVMFamily [](start = 24, length = 21)

just Quotas_List ?

**Comment by sanjaiganesh** [comment]
Adding ArmFeedbackRequested as there is one comment that is not responded to (or addressed) .. on adding this under preview api first.. also another minor comment I have added on operation name

**Comment by shahabhijeet** [comment]
@leizhaoPS all jobs are failing and this PR has [Do Not Merge]
Are you still thinking of merging this PR? if yes please make sure the CI is green.

**Comment by yungezz** [comment]
hi @leizhaoPS are you still on this PR? could you pls address comments and fix CI?

**Comment by leizhaoPS** [comment]
Sorry, the purpose of this PR is to get design sign-off from ARM, for now I close it and will check-in a new version later

**Pull request closed without merging**
